monai/handlers/checkpoint_saver.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from ignite.engine import Events
+from ignite.handlers import ModelCheckpoint
+
+
+class CheckpointSaver:
+    """
+    CheckpointSaver acts as an Ignite handler to save checkpoint data into files.
+    It supports to save according to metrics result, epoch number, iteration number
+    and last model or exception.
+
+    Args:
+        save_dir (str): the target directory to save the checkpoints.
+        save_dict (dict): source objects that save to the checkpoint. examples::
+
+            {'network': net, 'optimizer': optimizer, 'engine', engine}
+
+        name (str): identifier of logging.logger to use, if None, defaulting to ``engine.logger``.
+        file_prefix (str): prefix for the filenames to which objects will be saved.
+        save_final (bool): whether to save checkpoint or session at final iteration or exception.
+        save_key_metric (bool): whether to save checkpoint or session according to key metric value.
+        key_metric_name (str): the name of key_metric in ignite metrics dictionary.
+            if None, use `engine.state.key_metric` instead.
+        key_metric_n_saved (int): save top N checkpoints or sessions sorted by the key metric.
+        epoch_save_interval (int): save checkpoint or session every N epochs.
+        epoch_n_saved (int): save latest N epochs' checkpoints or sessions, 'None' is to save all.
+        iteration_save_interval (int): save checkpoint or session every N iterations.
+        iteration_n_saved (int): save latest N iterations' checkpoints or sessions, 'None' is to save all.
+        convert_to_single_gpu (bool): if current model is trained by multi GPU and
+            want to save to single GPU model, need to save `object.module` instead.
+
+    Note:
+        CheckpointHandler can be used during training, validation or evaluation.
+        example of saved files:
+            checkpoint_iteration=400.pth
```

Comment:
File names with operators in them may be valid names in Linux but that's not the case in other OSs, plus shells like bash really don't like them.

Reply:
Hi @ericspod ,

Thanks for your suggestion, the "=" is added by ignite:
https://github.com/pytorch/ignite/blob/master/ignite/handlers/checkpoint.py#L245
Maybe let's raise an issue to ignite?
Thanks.

Reply:
So it is, I think it should be mentioned. I had been using `global_step_transform` and a unscored saving scheme so I haven't seen this come up.

Reply:
OK, I will raise an issue to ignite soon.
Thanks.

Reply:
raised issue to ignite: https://github.com/pytorch/ignite/issues/1074
Thanks.